src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,36 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        final LocalDate dateLast = LocalDate.parse(dateTo, formatter);
+        final LocalDate dateFirst = LocalDate.parse(dateFrom, formatter);
+        StringBuilder sb = new StringBuilder();
+        int[] income = new int[names.length];
+        String[] temp = new String[4];
+        sb.append("Report for period " + dateFrom + " - " + dateTo + System.lineSeparator());
+        for (int j = 0; j < names.length; j++) {
+            for (int i = 0; i < data.length; i++) {
+                temp = data[i].split(" ");
+                if ((temp[1].equals(names[j]))
+                        && (LocalDate.parse(temp[0], formatter).isAfter(dateFirst)
+                        || LocalDate.parse(temp[0], formatter).isEqual(dateFirst))
+                        && (LocalDate.parse(temp[0], formatter).isBefore(dateLast)
+                        || LocalDate.parse(temp[0], formatter).isEqual(dateLast))) {
+                    income[j] += Integer.parseInt(temp[2]) * Integer.parseInt(temp[3]);
+                }
+            }
+            sb.append(names[j] + " - " + income[j]);
+            if (j == names.length - 1) {
```

Comment:
![image](https://user-images.githubusercontent.com/80981426/210428059-d78f3760-1460-433a-8514-34fc989b89cd.png)

Reply:
Created index constants for values in data: 
![image](https://user-images.githubusercontent.com/105455921/210654675-0fdf3258-8411-4c04-9464-9e8175307f1f.png)